### PR TITLE
[REF] base: Add vacuum method to clean ir.model.data xml_id '__export__'

### PR DESCRIPTION
**Pull request by moylop260**
These kinds of records could be too much during the time (4M for our oldest customers)
Many of them are useless after a while

It is possible thanks to:
 - https://github.com/odoo/odoo/pull/50661

Fix https://github.com/odoo/odoo/issues/35616

### CODE REVIEW

File: odoo/addons/base/models/ir_model.py [hunk 1]
```diff
@@ -2185,6 +2188,26 @@ def toggle_noupdate(self, model, res_id):
...
+        These kind of records could be too much during the time and many of them are useless after a while
+         - Just deleting records that are not used anymore after a while (VACUUM__EXPORT__MAX_WEEKS)
+         - Just deleting a few records each time (VACUUM__EXPORT__MAX_RECORDS)
+           in order to avoid raising an error if the worker timeout configuration is very low
+        """
+        limit_date = datetime.datetime.utcnow() - datetime.timedelta(weeks=VACUUM__EXPORT__MAX_WEEKS)
+
+        records = self.env['ir.model.data'].search([
+            ('module', '=', '__export__'), ('noupdate', '=', False),
+            ('create_date', '<', limit_date), ('write_date', '<', limit_date)],
```
Comment (luisg123v) [inline odoo/addons/base/models/ir_model.py:2203]: Why checking the creation date?

**Review by luisg123v — CHANGES_REQUESTED** [review]

File: odoo/addons/base/models/ir_model.py [hunk 2]
```diff
@@ -2185,6 +2188,26 @@ def toggle_noupdate(self, model, res_id):
...
 
+    @api.autovacuum
+    def _gc__export__xml_ids(self):
+        """Garbage collection in ir.model.data models only for xml_id '__export__'
+        These kind of records could be too much during the time and many of them are useless after a while
+         - Just deleting records that are not used anymore after a while (VACUUM__EXPORT__MAX_WEEKS)
+         - Just deleting a few records each time (VACUUM__EXPORT__MAX_RECORDS)
+           in order to avoid raising an error if the worker timeout configuration is very low
+        """
+        limit_date = datetime.datetime.utcnow() - datetime.timedelta(weeks=VACUUM__EXPORT__MAX_WEEKS)
```
Comment (fmdl) [inline odoo/addons/base/models/ir_model.py:2199]: `limit_date = fields.Datetime.now() - datetime.timedelta(weeks=VACUUM__EXPORT__MAX_WEEKS)`

File: odoo/addons/base/models/ir_model.py (hunk 1 above)
Comment (moylop260) [reply odoo/addons/base/models/ir_model.py:2203]: It was inspired by 
 - https://github.com/odoo/odoo/blob/c5d3a109f530cd0763a1c3860086b5b7abef1b64/addons/mail/wizard/mail_compose_message.py#L533-L534

`write_date` is enough for me, maybe a double check with `create_date` is better for Odoo dev team

**Comment by moylop260** [comment]
@xmo-odoo 
What do you think?

**Comment by moylop260** [comment]
@xmo-odoo 
Do you have any news about?

**Comment by moylop260** [comment]
@xmo-odoo 
Friendly reminder to give me your opinion about this

**Pull request closed without merging**